crates/cranelift/src/compiler/component.rs

```diff
@@ -799,7 +951,7 @@ impl<'a> TrampolineCompiler<'a> {
         self.raise_if_host_trapped(succeeded);
     }
 
-    fn raise_if_resource_trapped(&mut self, ret: ir::Value) -> ir::Value {
+    fn raise_if_i32_trapped(&mut self, ret: ir::Value) -> ir::Value {
```

Comment:
Perhaps something like `raise_if_negative_one`? It looks a bit odd above that an i64 return value is passed into a method that says `if_i32`.